internal/service/flexcluster/data_source.go

```diff
@@ -0,0 +1,55 @@
+//nolint:gocritic
+package flexcluster
+
+import (
+	"context"
+
+	"github.com/hashicorp/terraform-plugin-framework/datasource"
+	// "github.com/hashicorp/terraform-plugin-framework/datasource/schema"
+	// "github.com/hashicorp/terraform-plugin-framework/types"
+	"github.com/mongodb/terraform-provider-mongodbatlas/internal/config"
+)
+
+var _ datasource.DataSource = &ds{}
+var _ datasource.DataSourceWithConfigure = &ds{}
+
+func DataSource() datasource.DataSource {
+	return &ds{
+		DSCommon: config.DSCommon{
+			DataSourceName: resourceName,
+		},
+	}
+}
+
+type ds struct {
+	config.DSCommon
+}
+
+func (d *ds) Schema(ctx context.Context, req datasource.SchemaRequest, resp *datasource.SchemaResponse) {
+	// TODO: Schema and model must be defined in data_source_schema.go. Details on scaffolding this file found in contributing/development-best-practices.md under "Scaffolding Schema and Model Definitions"
+	resp.Schema = DataSourceSchema(ctx)
```

Comment:
you need to add `conversion.UpdateSchemaDescription(&resp.Schema)` after this line to add the Description field to all attributes like in the resource

Reply:
Noted! Since this PR is focused on implementing the schemas, I'll exclude this from this PR and include it in the data source implementation ticket 👍